### PR TITLE
store-gateway: stop relying on fixed number of samples per chunk

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
This is a prerequisite to #5927

Number of samples per chunk is changing in prometheus. Currently, only for native histograms, but also regular XOR chunks are capped at 1KiB.

This PR changes some tests to not rely on a hardcoded number of samples and also updates/removes some constants in the code. Instead it queries the blocks on disk using Prometheus code and compares this to what the store-gateway returns.

For chunk estimations I changed the tests to assert that the estimation is not off by more than 10% from the real number of chunks. I'd like to get a review from @charleskorn on whether this is an appropriate approach.

### CODE REVIEW

File: pkg/storegateway/bucket_test.go
Comment (charleskorn) [inline pkg/storegateway/bucket_test.go:2264]: Could we compare `estimatedChunks` to the number of chunks in `seriesSet[0].Chunks`? Or is there a reason why we need to read the chunks from disk?

Comment (dimitarvdimitrov) [reply pkg/storegateway/bucket_test.go:2264]: good point. I don't think there's a reason to read from disk

Comment (dimitarvdimitrov) [reply pkg/storegateway/bucket_test.go]: did that in 1277b1590

**Review by charleskorn — APPROVED** [review]
Changes LGTM, but I'm no TSDB expert - might be worth getting a review from someone else as well.

**Comment by dimitarvdimitrov** [comment]
I'll merge this to unblock #5927; happy to address comments after the merge, but I think the tsdb-related changes are fairly trivial, it's just using the BlockReader interface

**Pull request merged**
